pkg/controller/elasticsearch/user/roles.go

```diff
@@ -66,6 +69,51 @@ var (
 	PredefinedRoles = RolesFileContent{
 		ProbeUserRole:     esclient.Role{Cluster: []string{"monitor"}},
 		ClusterManageRole: esclient.Role{Cluster: []string{"manage"}},
+		DiagnosticsUserRole: esclient.Role{
+			Cluster: []string{"monitor", "monitor_snapshot", "manage", "read_ilm", "read_security"},
+			Indices: []esclient.IndexRole{
+				{
+					Names:                  []string{"*"},
+					Privileges:             []string{"monitor", "read", "view_index_metadata"},
+					AllowRestrictedIndices: ptr.To[bool](true),
+				},
+			},
+			Applications: []esclient.ApplicationRole{
+				{
+					Application: "kibana-.kibana",
+					Resources:   []string{"*"},
+					// Unfortunately the following privileges are not sufficient access all Kibana APIs
+					// that are required for the ECK diagnostics. In the future we should try again to
+					// generate a more fine-grained role and not use "*".
+					Privileges: []string{
+						"*",
```

Comment:
Are you still investigating this or is this the least privileges we can give?

Reply:
@pebrc update: I've found that some of these endpoints require platinum licenses, which is why I'm getting unauthorized errors when attempting to call them, even with privileges that should allow me to call them, as I'm running a dev version of ECK `make go-run` which doesn't support enterprise features.

I'm going to try and test this a bit differently and update.

Reply:
Ok @pebrc with the current roles, which are as limited as I can get them, I can successfully run eck-diagnostics with no issues. This should be ready for final reviews. An upcoming PR to eck-diagnostics will be upcoming once this is merged.